drv/sprot-api/src/error.rs

```diff
@@ -28,6 +29,7 @@ pub enum SprotError {
     Spi(SpiError),
     Update(UpdateError),
     Sprockets(SprocketsError),
+    Attest(AttestError),
```

Comment:
You shouldn't add an AttestError here, since it's not directly sprot related. In reality, the `Sprockets` and `Update` errors probably shouldn't live here either but that ship has sailed. 

We figured this out later on with both dump and caboose APIs. We deal with errors that occur on the RoT and must transit sprot by encoding them directly into the underlying response type. So `RspBody::Attest` should contain a `Result` that has an `AttestError` variant and then the sprot api should return a `SprotOrAttest` error. See how the `Caboose` does it for inspiration.

Reply:
That makes a lot of sense. I hadn't really picked up on it till now but the `Caboose` errors are conspicuously absent here. I'll update my approach to follow the `Caboose`.

Reply:
PR has been updated.